components/org.wso2.scim2.testsuite.core/src/main/java/org/wso2/scim2/testsuite/core/tests/BulkTestImpl.java

```diff
@@ -0,0 +1,1643 @@
+                for (Header header : headers) {
+                    headerString.append(String.format("%s : %s \n", header.getName(), header.getValue()));
+                }
+                responseStatus = response.getStatusLine().getStatusCode() + " " +
+                        response.getStatusLine().getReasonPhrase();
+                //  Get the created user locations.
+                resourceStatusCodes = getStatus(responseString);
+                //  Get the created user locations.
+                createdResourceLocations = getLocations(responseString);
+            } catch (Exception e) {
+                /*
+                 Read the response body.
+                 Get all headers.
+                 */
+                Header[] headers = response.getAllHeaders();
+                for (Header header : headers) {
+                    headerString.append(String.format("%s : %s \n", header.getName(), header.getValue()));
+                }
+                responseStatus = response.getStatusLine().getStatusCode() + " "
+                        + response.getStatusLine().getReasonPhrase();
+                if (requestPaths[i].getTestSupported() &&
+                        !requestPaths[i].getTestCaseName().equals("Bulk post operation without bulk Id") &&
+                        !requestPaths[i].getTestCaseName().equals("Bulk post operation without path") &&
+                        !requestPaths[i].getTestCaseName().equals("Bulk post operation without data") &&
+                        !requestPaths[i].getTestCaseName().equals("Bulk post operation with fail on errors")) {
+                    // Check for status returned.
+                    subTests.add(ComplianceConstants.TestConstants.STATUS_CODE);
+                    subTests.add("Actual : " + response.getStatusLine().getStatusCode());
+                    subTests.add("Expected : 200");
+                    subTests.add("Status : Failed");
+                    subTests.add(StringUtils.EMPTY);
+                    long stopTime = System.currentTimeMillis();
+                    testResults.add(new TestResult(TestResult.ERROR, requestPaths[i].getTestCaseName(),
+                            "Could not perform bulk request at " + url,
+                            ComplianceUtils.getWire(method, responseString, headerString.toString(), responseStatus,
+                                    subTests), stopTime - startTime));
+                    continue;
+                }
+            }
+            Boolean pass = false;
+            for (Integer status : resourceStatusCodes) {
+                if (status == 201) {
+                    // Check for status returned.
+                    subTests.add("Check each resource status");
+                    subTests.add("Actual : " + status);
+                    subTests.add("Expected : 201");
+                    subTests.add("Status : Success");
+                    subTests.add(StringUtils.EMPTY);
```

Comment:
Define a common util method for this section (line 650-653) and reuse it. Define `:` as contact with a name similar to  `ASSERTION_VALUE_SEPARATOR`